unix/xserver/hw/vnc/xvnc.c

```diff
@@ -277,8 +278,11 @@ ddxProcessArgument(int argc, char *argv[], int i)
         vncInitRFB();
     }
 
-    if (argv[i][0] == ':')
+    if (argv[i][0] == ':') {
         displaySpecified = TRUE;
+        strncpy(displayStr, argv[i], 31);
+        displayStr[31] = '\0';
+    }
```

Comment:
Looks like `display` is the canonical truth here. Probably best to look at that once everything is set up.